java/src/main/java/org/rocksdb/MutableOptionValue.java

```diff
@@ -17,7 +17,7 @@
       extends MutableOptionValue<T> {
     protected final T value;
 
-    private MutableOptionValueObject(final T value) {
+    protected MutableOptionValueObject(final T value) {
```

Comment:
Everything looks good, but I am not sure why you made the visibility change here?